cli/src/main.rs

```diff
@@ -1,35 +1,91 @@
+use std::env;
 use std::{fs, path::PathBuf};
 
 use argh::FromArgs;
-use wasmstation::backend::WasmerBackend;
-use wasmstation::console::Console;
+use wasmstation::{backend::WasmerBackend, console::Console};
 
 #[derive(FromArgs)]
 #[argh(description = "Run wasm4 compatible games.")]
 struct Args {
-    #[argh(positional)]
-    path: PathBuf,
-    #[argh(
-        option,
-        short = 's',
-        default = "3",
-        description = "scale factor for the window"
-    )]
-    display_scale: u32,
+    #[argh(subcommand)]
+    subcommand: Subcommand,
+}
+
+#[derive(FromArgs)]
+#[argh(subcommand)]
+enum Subcommand {
+    Run(Run),
+    Create(Create),
 }
 
 fn main() {
     let args: Args = argh::from_env();
 
     pretty_env_logger::init();
 
+    match args.subcommand {
+        Subcommand::Run(args) => run(args),
+        Subcommand::Create(args) => create(args),
+    }
+}
+
+/// Run a WASM-4 game.
+#[derive(FromArgs)]
+#[argh(subcommand, name = "run")]
+struct Run {
+    #[argh(positional)]
+    path: PathBuf,
+    /// default scale factor for the window
+    #[argh(option, short = 's', default = "3")]
+    display_scale: u32,
+}
+
+fn run(args: Run) {
     let wasm_bytes = fs::read(&args.path).expect("failed to read game");
-    let console = Console::new();
 
     wasmstation::launch(
-        WasmerBackend::new(&wasm_bytes, &console).unwrap(),
+        WasmerBackend::new(&wasm_bytes, &Console::new()).unwrap(),
         &args.path,
         args.display_scale,
     )
     .unwrap();
 }
+
+/// Create an executable from a WASM-4 game.
+#[derive(FromArgs)]
+#[argh(subcommand, name = "create")]
+struct Create {
+    #[argh(positional)]
+    path: PathBuf,
+    /// scale factor of the window
+    #[argh(option, short = 's', default = "3")]
+    display_scale: u32,
+    /// name of the game
+    #[argh(option, short = 't', default = "\"cart\".to_string()")]
```

Comment:
defaulting to 'cart' is not very convenient. When I look the cartridge files on wasm4.org, I notice that they all have distinct names - so why not take the file name (with .wasm removed if the file has that extension) as the default game name?

Reply:
Ah this is another good point. I believe my original intention was to have the "name" argument be the title of the window, however the current implementation of `renderer::launch` creates the title of the window from the filename of the .wasm file (I plan to change this in the future to be more flexible).

In my most recent commit I completely removed the "name" argument from `wasmstation create`. Most often, people will want the Cargo project name to coincide with the executable name and the title on the window. In the most recent implementation all of these names come from the [`Path::file_stem`](https://doc.rust-lang.org/std/path/struct.Path.html#method.file_stem) of the input file.